src/networking.c

```diff
@@ -4690,10 +4809,33 @@ int closeClientOnOutputBufferLimitReached(client *c, int async) {
 void flushSlavesOutputBuffers(void) {
     listIter li;
     listNode *ln;
+    int revert_flags = 0;
 
     listRewind(server.slaves,&li);
     while((ln = listNext(&li))) {
         client *slave = listNodeValue(ln);
+
+        /* Fetch the replica clients that are currently running in IO thread.
+         * If shutdown fails, they will be returned back to IO thread in
+         * handleClientsWithPendingWrites after the repl backlog is fed with new
+         * data. */
+        revert_flags = 0;
+        if (slave->running_tid != IOTHREAD_MAIN_THREAD_ID) {
+            fetchClientFromIOThread(slave);
+            /* If the slave doesn't have any pending replies nothing to do 
+             * anyways. */
+            if (!clientHasPendingReplies(slave)) {
+                continue;
+            }
+
+            putClientInPendingWriteQueue(slave);
+
+            if (!(slave->io_flags & CLIENT_IO_WRITE_ENABLED)) {
+                slave->io_flags |= CLIENT_IO_WRITE_ENABLED;
+                revert_flags |= CLIENT_IO_WRITE_ENABLED;
```

Comment:
seems we don't need revert_flags, since we will check if it need to write reply in processClientsFromMainThread